chef/cookbooks/network/recipes/default.rb

```diff
@@ -398,32 +398,51 @@ def get_datapath_id_for_ovsbridge(bridge)
 
   # Do some MTU checks/configuration for the parent of the vlan nic
   if nic.is_a?(Nic::Vlan)
-    # 1) validate that the parent of vlan nic is not used by a network directly
-    # and the requested mtu for the parent is lower than the vlan nic mtu.
-    # That would not work and setting the mtu on the vlan would fail.
-    networks_using_parent = if_mapping.select { |net, ifaces| ifaces.include? nic.parent }.keys
-    networks_using_parent.each do |net_name|
-      net = Barclamp::Inventory.get_network_by_type(node, net_name)
-      unless net.use_vlan
-        nic_parent = Nic.new nic.parent
-        if nic_parent.mtu.to_i < ifs[nic.name]["mtu"].to_i
-          msg = "#{nic.name} wants mtu #{ifs[nic.name]["mtu"]} but network #{net_name} " \
-                "using the parent nic #{nic.parent} wants a lower mtu #{net.mtu}. " \
-                "This network mtu configuration is invalid."
+    if nic.parent
+      parent_nic = Nic.new(nic.parent)
+      if parent_nic.bond?
+        # parent is a bond
+        if parent_nic.key?("mtu") && parent_nic["mtu"] < ifs[nic.name]["mtu"].to_i
+          # parent has custom mtu lower than the nic custom mtu
+          msg = "Parent bond #{parent_nic.name} has a custom mtu of #{parent_nic["mtu"]} "\
+                "but net #{nic.name} wants a mtu of #{nic["mtu"]}."\
+                "This configuration is invalid."
           Chef::Log.fatal(msg)
           raise msg
         end
+        # check bond slaves interfaces for mtu values
```

Comment:
Is this really needed? Under what circumstance can the bond slaves have a diffferent MTU than the bond itself?